src/components/ClassInfoCard.tsx

```diff
@@ -39,6 +39,9 @@ export const ClassInfoCard: React.FC<ClassInfoProps> = ({
     const { t } = useTranslation();
     const { me } = useMe();
 
+    const fullClass =
+        me && me.role === roles.PARENT && cardInfo.spaceAvailable < 1;
+
```

Comment:
hmm, although it's a bit ugly, I would think passing a new prop like isFull from class list -> class info card etc would be better then duplicating  this logic in your two `ClassInfoCard.tsx` and `ClassInfoModal.tsx`